### ISSUE TITLE
Stream freezes if enough of its chunks fail to start

### ISSUE THREAD

**Issue by vcschapp** [gogama/incite#15]
# Background

This is related to the bug [I just fixed](https://github.com/gogama/incite/commit/8abbdc5ad29cad358d40068223e549016416d50a), in which, after the v1.3.0 concurrency refactor, the query manager in `mgr.go` was not correctly respecting its own `Config.Parallel` configuration. This could result in the query manager trying to start more queries than there is available service quota in CWL, resulting in transient errors like:

```
LimitExceededException: Account maximum query concurrency limit of
    [10] reached. (Service: AWSLogs; Status Code: 400; Error Code:
    LimitExceededException; Request ID: ...
````

The query manager was correctly retrying these so you wouldn't even notice them unless you looked at the logs, but...

# Bugs

## Main Bug - starter

Relatedly, the chunk start worker (`starter.go`) has a bug where if it gets a transient failure to start a query, [it doesn't set the chunk error](https://github.com/gogama/incite/blob/main/starter.go#L56-L61).

The result is if the chunk retry limit gets exceeded within the starter, the chunk goes back to the query manager without any error being set in it. Because the chunk state is still `starting`, the manager does the right thing and goes through the [`killStream` flow](https://github.com/gogama/incite/blob/main/mgr.go#L377-L384), but as there is no error, `killStream` does not actually cause the stream to die, with the result that anyone blocked on a stream read will wait forever instead of getting an error message.

## Second Bug - poller

There is an analogous bug in the chunk poll worker (`poller.go`) [where it handles temporary errors](https://github.com/gogama/incite/blob/main/poller.go#L56-L59).

# Speculation on Fixing

To my mind the following fixes need to be done:

1. Slight restructure of start/poll manipulations so they always set `c.err` if there's an error, even if it's transient.
2. The worker loop should `nil` out the chunk error before [running the manipulation](https://github.com/gogama/incite/blob/e661272f2e8c07e15297f8e29d727bff9ffc439a/worker.go#L82).
3. Obviously unit tests should exercise this scenario both holistically, ideally via scenario, and for each worker.

**Comment by vcschapp** [comment]
The fix ended up being exactly as expected. I'm just testing and will push the commit shortly.

**Comment by vcschapp** [comment]
This is done as of commit [87514ee](https://github.com/gogama/incite/commit/87514ee1d43328756a76c36b8142995acc494c45) and released into [v1.3.1](https://github.com/gogama/incite/releases/tag/v1.3.1),

Tested and working with 50,000 full cycles of the unit tests on Windows 10 and 100,000 full cycles of the unit tests on AL2. There are some tantalizing signs that there is still a hard-to-find race condition, since unit tests on AL2 run with 1MM and 10MM cycles seem to have stalled out. I doubt that's a regression and defer it for another issue.
